pytorch_lightning/trainer/trainer.py

```diff
@@ -85,7 +85,7 @@ class Trainer(
     def __init__(
         self,
         logger: Union[LightningLoggerBase, Iterable[LightningLoggerBase], bool] = True,
-        checkpoint_callback: Union[ModelCheckpoint, bool] = True,
+        checkpoint_callback: bool = True,
```

Comment:
> Q: Why can we not just remove the checkpoint_callback argument all together?
> A: Lightning philosophy is to provide good defaults, this includes checkpointing. We need a way to turn it off though, i.e., checkpoint_callback=False

What about keeping the `checkpoint_callback` parameter but changing its default value to `False`? I think it will be pretty annoying to have to set `checkpoint_callback=False` every time you pass a custom `ModelCheckpoint` via callbacks. And I think most people use a custom `ModelCheckpoint` instead of just `checkpoint_callback=True`

Reply:
> What about keeping the checkpoint_callback parameter but changing its default value to False? 

doesn't solve the problem I'm trying to solve here, which is eliminate ambiguity when restoring the state of trainer. see answer of 2nd FAQ question.

> pretty annoying to have to set checkpoint_callback=False every time you pass a custom ModelCheckpoint

with this PR proposal, the value will be ignored if you pass in a custom one. False is only needed when you want to disable checkpointing completely. I believe I have this covered in a test.

Reply:
i agree with @carmocca , this is super confusing when adding my own checkpoint callback. given how loose the default checkpoint callback is, and with the coming customizations, I'd rather drop the `checkpoint_callback` arg altogether and force everything to be configured through `callbacks`. Given the implementation already exists, I personally don't think it's much of a request for people to create a callback (and confirm their settings while doing so) and pass it along

Reply:
yes fine with me, I don't have strong preference here. 
Note that this PR does NOT close the disussion on whether there should be a checkpoint_callback arg or not. 
I'm simply restricting what can be passed to the argument.